webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/OrderedScrambles.java

```diff
@@ -0,0 +1,118 @@
+                ArrayList<ArrayList<ScrambleRequest>> scrambleRequestListByDay = new ArrayList<ArrayList<ScrambleRequest>>();
+                
+                String roomName = wh.getRoomName(room);
+                                
+                for (JsonElement activity : wh.getActivities(room)) {
+                    String activityCode = wh.getActivityCode(activity);
+
+                    for (ScrambleRequest scrambleRequest : wh.getScrambleRequests(activityCode)) {
+
+                        DateTime activityStartTime = wh.getActivityStartTime(activity, timezone);
+
+                        int activityDay = Days.daysBetween(competitionStartDate.withTimeAtStartOfDay(), activityStartTime.withTimeAtStartOfDay()).getDays()+1;
+                        
+                        if (!dayList.contains(activityDay)) {
+                            dayList.add(activityDay);
+                            scrambleRequestListByDay.add(new ArrayList<ScrambleRequest>());
+                        }
+
+                        int index = dayList.indexOf(activityDay);
+                        scrambleRequest.roundStartTime = activityStartTime;
+                        scrambleRequestListByDay.get(index).add(scrambleRequest);
+                    }
+                }
+
+                for (int index = 0; index<dayList.size(); index++) {
+                    if (scrambleRequestListByDay.get(index).size()>0) {
+                        
+                        Collections.sort(scrambleRequestListByDay.get(index));
+                        
+                        String pdfFileName = "Printing/Ordered Scrambles/";
+
+                        if (hasMultipleVenues) {
+                            pdfFileName += venueName.replace('/', ' ')+"/";
+                        }
+
+                        if (hasMultipleDays || dayList.size() > 1) { // Double check, just in case.
+                            pdfFileName += "Day "+dayList.get(index)+"/";
+                        }
+
+                        pdfFileName += "Ordered Scrambles";
+                        
+                        // In addition to different folders, we stamp venue, day and room in the PDF's name
+                        // to prevent different files with the same name.
+                        if (hasMultipleVenues) {
+                            pdfFileName += " - " + venueName.replace('/', ' ');
+                        }
+                        if (hasMultipleDays || dayList.size() > 1) {
+                            pdfFileName += " - Day "+dayList.get(index);
+                        }
+                        
+                        if (hasMultipleRooms) {
+                            pdfFileName += " - "+roomName.replace('/', ' ');
+                        }
+                        pdfFileName += ".pdf";
+                        
+                        // removing slashes may add double+ extra spaces
+                        pdfFileName = pdfFileName.replaceAll("\\s+", " ");
```

Comment:
(related to all my comments above about using `toFileSafeString`) This feels like reasonable logic to me. If you want to keep it around, could you add it to `toFileSafeString` instead?